prboom2/src/CMakeLists.txt

```diff
@@ -517,50 +517,48 @@ function(AddGameExecutable TARGET SOURCES)
         ${CMAKE_BINARY_DIR}
         ${CMAKE_CURRENT_SOURCE_DIR}
     )
+
     target_link_libraries(${TARGET} PRIVATE
         ${SDL2_LIBRARIES}
+        SDL2_mixer::SDL2_mixer
+        ZLIB::ZLIB
     )
+
     if(WIN32)
         target_link_libraries(${TARGET} PRIVATE
             winmm
             comctl32
         )
     endif()
+
     set_target_properties(${TARGET} PROPERTIES
         RUNTIME_OUTPUT_DIRECTORY ${PRBOOM_OUTPUT_PATH}
     )
-    if(SDL2_IMAGE_FOUND)
-        target_include_directories(${TARGET} PRIVATE ${SDL2_IMAGE_INCLUDE_DIRS})
```

Comment:
Could you tell me what the include_directories lines did and why they aren't needed anymore?

Reply:
Previously, Find modules would declare variables `*_INCLUDE_DIRS` and `*_LIBRARIES` that contained literal paths to, respectively, the include directories and libraries.

Now, Find modules create imported CMake targets (`add_library(Foo::Foo UNKNOWN IMPORTED)`) which contain both the library itself, its dependencies and include directories:
```cmake
set_target_properties(Foo::Foo PROPERTIES
  IMPORTED_LOCATION "/path/to/the/library"
  INTERFACE_INCLUDE_DIRECTORIES "/path/to/include"
  INTERFACE_LINK_LIBRARIES "list;of;other;dependencies")
```

Linking to this target through `target_link_libraries` applies all these properties to the target, making `target_include_directories` redundant.

The only exception to that is with SDL2. On 2.0.10 and below, when SDL is built with autotools, the CMake config is really bare, only providing `SDL2_LIBRARIES` and `SDL2_INCLUDE_DIRS` and no target. So they are used to keep compatibility with 2.0.7.

Reply:
Thanks, makes sense 👍